application/src/main/java/org/opentripplanner/model/TripTimeOnDate.java

```diff
@@ -371,4 +390,76 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(tripTimes, stopIndex, tripPattern, serviceDate, midnight);
   }
+
+  public TripPattern pattern() {
+    return tripPattern;
+  }
+
+  /**
+   * Returns the previous stop times in the trip. If it's the first stop in the trip, it returns an
+   * empty list.
+   */
+  public List<TripTimeOnDate> previousTimes() {
```

Comment:
This exceeds the responsibility of the class - and exploit the design. A singel instance of something should normally not now about its neighbours. Would it be ok to convert these into static factory methods instead? Two alternatives are to create a collection `TripTimeOnDateList` (normaly the name would be `TripTimeOnDates - but that is confusing`) and/or `TripTimeOnDateFactory`.

Reply:
In light of what we talked about in the dev meeting (this being as DTO and not a domain model) do you still think I should move these methods to a factory?